TextToSpeech/impl/TTSDownloader.cpp

```diff
@@ -57,77 +57,71 @@ void TTSDownloader::download(TTSConfiguration config)
 
 void TTSDownloader::downloadThread()
 {
-   while(active)
-   {	   
-       std::unique_lock<std::mutex> lock(m_queueMutex);
-       m_condition.wait(lock,[this]{return needDownload.load();});
+    while(active)
+    {
+        std::unique_lock<std::mutex> lock(m_queueMutex);
+        m_condition.wait(lock,[this]{return needDownload.load();});
 
-       if(active == false)
-       {
-           //invoked from destructor..no download
-           break;
-       }
-       needDownload = false;
-       std::string tts_request;
-       TTSURLConstructer url;
-       m_objectMutex.lock();
-       tts_request = url.constructURL(m_config,"",true);
-       m_objectMutex.unlock();
-       while(tts_request.empty())
```

Comment:
are we not concerned about this case anymore?

Reply:
not required, in case of failure due no internet/server down will be retried on the next iteration.